api/wasm/indigo-ketcher/indigo-ketcher.cpp

```diff
@@ -362,14 +363,12 @@ namespace indigo
         bool isFirst = true;
         for (const auto& str : vs.vec)
         {
-            if (vs.vec.size() > 1)
-            {
-                if (isFirst)
-                    isFirst = false;
-                else
-                    ss << "+";
+            if (isFirst)
+                isFirst = false;
+            else
+                ss << "+";
+            if( vs.vec.size() > 1 || vs.both )
```

Comment:
This expression is invariant in the cycle and can be calculated outside `for`.
Moreover, the only case when it's false (when size == 1) does not require cycle `for` at all.
So,
```
if (vs.vec.size() == 1 && !vs.both)
{
   ss << str;
   return;
}
```
// otherwise for

Reply:
fixed